lib/ansible/modules/cloud/amazon/codecommit_ssh_key_facts.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+
+# -*- coding: utf-8 -*-
+#
+# (c) 2017, Pat Sharkey <psharkey@cleo.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: codecommit_ssh_key_facts
+short_description: Get the AWS SSH Public Key facts
+description:
+    - Get the AWS CodeCommit SSH Public Key facts for a given user
+version_added: "2.4"
+author: "Pat Sharkey, (@psharkey)"
+options:
+  user_name:
+    description:
+      - The name of the user whose SSH public key will be returned
+    required: true
+  encoding:
+    description:
+      - The desired format for the returned key PEM | SSH
```

Comment:
@gundalow Thanks for your comments. Regarding the name, AWS has a CodeCommit service (http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_ssh-keys.html) and this module will gather the SSH public key(s) for all IAM users.  Do you have any suggestions for a different name - maybe `codecommit_ssh_facts` or `code_commit_facts` (with the caveat that only SSH information is currently supported)?
I'll work on adding `PEM` and `SSH` as the only valid choices to the encoding parameter. This is simply passed as a required parameter to the boto method `get_ssh_public_key` (http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.get_ssh_public_key).